lib/AppInfo/Application.php

```diff
@@ -30,13 +30,17 @@
 use OCP\AppFramework\Bootstrap\IBootContext;
 use OCP\AppFramework\Bootstrap\IBootstrap;
 use OCP\AppFramework\Bootstrap\IRegistrationContext;
+use Office365\Runtime\Auth\AuthenticationContext;
+use Office365\Runtime\Auth\SamlTokenProvider;
```

Comment:
Seeing both paths being in the same namespace, does it make sense to allow giving class names only or even namespaces only?

Reply:
Both might be too broad in other cases.

Reply:
Well that's why I meant "Allow", but yeah then let's keep it like this.
I just fear the moment they change an internal method name or something and it logs again

Reply:
> Well that's why I meant "Allow", but yeah then let's keep it like this. I just fear the moment they change an internal method name or something and it logs again

Absolutely. But that can happen to namespaces as well as class names (also happened in this lib in the past).

The only sustainable solution will arrive with PHP 8.2's sensitive parameter support.

Reply:
@nickvergessen added an integrattion test that should act as safeguard nevertheless